faucet.js

```diff
@@ -66,20 +72,33 @@ class Faucet {
      * @param dapp The address of dapp 
      */
     async checkAppliable(dapp) {
-        let r, sponsorInfo, faucetParams;
+        if(dapp === null) return {flag: false, message:''};
+        let r, sponsorInfo, faucetParams, collateralForStorage, oldDetail;
         try {
-            faucetParams = await this.getFaucetParams();
+            faucetParams = await this.getFaucetParams(dapp);
             sponsorInfo = await this.cfx.getSponsorInfo(dapp);
-            if(sponsorInfo.sponsorForCollateral !== this.address && sponsorInfo.sponsorBalanceForCollateral > faucetParams.collateral_bound) {        
+            collateralForStorage = await this.cfx.getCollateralForStorage(dapp);
+            
+            if(sponsorInfo.sponsorForCollateral === this.lastAddress) {
```

Comment:
Space missed after `if` keyword, strongly recommend to pay attention to your code style or use tool to format automatically. You're developing a product, not writing a demo.

Reply:
Code formated